src/NUnitTestAdapter/AdapterSettings.cs

```diff
@@ -368,6 +358,23 @@ string ValidatedPath(string path, string purpose)
             }
         }
 
+        private void MapTestCategory(string vsTestCategoryType)
+        {
+            switch (vsTestCategoryType.ToLower())
```

Comment:
The idiomatic thing to do is `string.Equals(x, y, StringComparer.OrdinalIgnoreCase)`. ToLower has issues with certain culture settings and it produces an unnecessary string allocation.

Reply:
Recommended:

```cs
if (Enum.TryParse<VsTestCategoryType>(vsTestCategoryType, ignoreCase: true, out var value))
{
    VsTestCategoryType = value;
}
else
{
    _logger.Warning(
        $"Invalid value ({vsTestCategoryType}) for VsTestCategoryType. Valid values are NUnit and MSTest.");
}
```

Reply:
Thanks!

Reply:
We're still on .net 3.5, where this doesn't exist.-......   We really need to get off 3.5

Reply:
Okay, now ToLower is still not equivalent to StringComparer.OrdinalIgnoreCase. `"NUNIT".ToLower()` is not `"nunit"` if the user's locale is set to Turkish.

Reply:
Options I'd recommend:

- Add a utility `TryParseEnum<T>` method by catching ArgumentException from Enum.Parse (I do this on .NET 3.5)

- Convert `switch` to `if` and use `nameof(VsTestCategoryType.NUnit).Equals(vsTestCategoryType, StringComparison.OrdinalIgnoreCase)`

- Use a private static readonly `new Dictionary<string, T>(StringComparer.OrdinalIgnoreCase)` filled at runtime from the enum

Reply:
Done, but seems a waste - we should get over to 4.X asap